### ISSUE TITLE
Some commands not showing up and inaccurate command duration

### ISSUE THREAD

**Issue by gvlassis** [atuinsh/atuin#1003]
I am using freshly installed Atuin v14.0.1 through Homebrew on bash+macOS.

I have encounterd two weird issues that might or might not be related:
1) Sometimes a command does not show up in history (see "fastfetch" command in first screenshot). I have not noticed any pattern, even though there might be one.
2) Sometimes the duration of a command is inaccurate (see "which atuin" in second screenshot). Also, it sometimes shows up as 0 on the first invocation of the interactive history search and then as something else on the following invocations (compare "which atuin" in first and second screenshots). 
![Screenshot 2023-05-23 at 12 49 22 PM](https://github.com/ellie/atuin/assets/74119653/69a5a61a-92be-46c4-b0c0-6c5d8947e248)
![Screenshot 2023-05-23 at 12 49 49 PM](https://github.com/ellie/atuin/assets/74119653/10f83f4c-3be4-496d-8273-fa77a6129683)

**Comment by conradludgate** [comment]
Hmm. That's very weird. What shell do you use?

**Comment by danwilliams** [comment]
I have a brand-new installation on Bash, and I am finding that lots of my commands do not end up in history. Very odd.

I will run a command, hit up arrow to run it again - and it's not there. I have up arrow disabled for Atuin at present, too, but the command also doesn't show int he Atuin interface with Ctrl-R.

There is no pattern to this and if I run it again the command will generally then show up.

The problem is it leads to very unreliable behaviour and slows down shell interaction as you have to check what's about to be edited and not rely on your memory.

**Comment by danwilliams** [comment]
I think I have found by usage an observation that the problem happens consistently when using Ctrl-R - the command chosen by that process never seems to be added afresh to the current history context, meaning that it's not available when you press Up Arrow. I'm not sure if it happens in other ways as well, but I will continue to observe.

**Comment by arcuru** [comment]
@danwilliams You haven't said, but I suspect that you are using [`enter_accept`](https://atuin.sh/docs/config/#enter_accept) and you are using Atuin v17.0.0.

v17.0.1 has #1342, which will ensure that the command you select using `enter_accept` from the Atuin interactive search will show up in the atuin history.

I've opened #1383 to make sure that those commands will be added to the bash history as well so that your up-arrow usage will also work.

You can use both these fixes today by putting the code of `atuin init bash --disable-up-arrow` into .bashrc and editing the `__atuin_history` function to the version in the #1383 PR.

**Comment by danwilliams** [comment]
@arcuru thanks for the super-fast and extremely helpful response 🙂 You're right, I should have mentioned the version. I simply did `cargo install atuin` and have checked now - I'm on v17.0.1, which is interesting. I've checked the manual so that I understand what `enter_accept` means and yes you are correct, I am using that.

I am not sure if #1342 is working okay, because I'm meant to have that fix and yet it seems commands are not showing up in Atuin's history. I will keep an eye on that, as it's the up-arrow behaviour that is the most noticeable. I'm not sure if I've seen them missing from Atuin's history since applying `--disable-up-arrow`.

Thank you for opening #1383 - I am already using `atuin init bash --disable-up-arrow` and that works fine (I find the Atuin up-arrow behaviour very odd and there's seemingly no way to restrict it to a terminal session, so I disabled it). I don't know if/where I can edit the atuin.bash file under a normal installation, so I cloned the repository and installed from source with your fix from the PR. I can confirm that doing that instantly fixes the up-arrow problem ❤️ 

I will keep an eye on the Atuin history issue to see if I can confirm that either way.

Thanks again!

**Comment by danwilliams** [comment]
Yup, unfortunately #1342 is not fully fixed, as some commands are definitely not getting added to history. #1383 is indeed fixed though, and has not been a further problem.

**Comment by danwilliams** [comment]
I've noticed two types of command that don't show up in Atuin's log:

  - Compound commands, e.g. `cargo build && cargo clippy && cargo doc`
  - Commands that use `git filter-branch` - I have no idea why that would be the case; I have two that I've tried and one shows up whereas one doesn't. Both are accessed via short names so are not long complex commands.  I'm trying to work out more to report on this.

**Comment by danwilliams** [comment]
Interactive commands don't seem to get added - e.g. `nano`. (Talking about the Atuin history here.)

**Comment by cenuij** [comment]
> Yup, unfortunately #1342 is not fully fixed, as some newly-typed commands are definitely not getting added to Atuin's history (they are in Bash history and I can access them through up-arrow).

In my experience, whenever I use Ctrl-R and then ESC (i.e. I don't select anything), whatever command I then type on the command line does not get added to the history. However, if I do Ctrl-R, ESC and then ENTER (i.e blank command) before I type a new command, it works.

Also, when I use Ctrl-R and then TAB to edit the command, the edited command is also not added. I think it should be added?
